src/components/TextInputField.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as React from "react";
+
+export interface TextInputFieldProps {
+    label: string;
+    value: string;
+    onChange: (newValue: string) => void;
+}
+
+export class TextInputField extends React.PureComponent<TextInputFieldProps> {
+    /**
+     * The factory this component uses to render itself. Set to a different value to override.
+     * @param props The component properties
+     * @returns The component, rendered.
+     */
+    public static renderFactory = (props: TextInputFieldProps): React.ReactNode => (
```

Comment:
This pattern will be very easy to forget about when refactoring, it is very strange

could we not move the module-api up to element-web and then module-api can depend on react-sdk and use components like Spinner etc inherently?

Reply:
We're actively avoiding a react SDK dependency because the modules themselves cannot and should not have easy access to that dependency.

Reply:
The layer provides two main benefits:

1. It gives us something to check version against implicitly, ensuring we're only including supported modules in our build.
2. It avoids modules depending on the react-sdk either directly or indirectly, reducing risk of developers breaking free of the module system and making bad decisions. The react-sdk is also not a tiny project, so it ultimately saves disk space on developer machines, and it reduces the chances of dependency conflicts when webpack tries to figure out what is going on.
  * A react-sdk dependency would also overly tie a module to an application version when in fact it could likely be supported across breaking changes of the react-sdk: this layer abstraction allows for a module to write code against a standard set of functions and be fine for several releases. Essentially, it allows us to break the app and module API surface separately, a benefit that projects like Synapse don't have.

Together, these are reason enough to warrant the new layer. While the layer itself is a bit annoying (requires someone to keep an eye on it), it's expected to be even less traffic than the widget-api or analytics-events dependencies.